### PR TITLE
Added proper deprecations for the Syslog and JsonFormatter changes.

### PR DESCRIPTION
**Pull request by rofinn**
These deprecations aren't great, but they should be good enough that people should be informed of the breaking changes at least.

### CODE REVIEW

File: src/Memento.jl [hunk 1]
```diff
@@ -38,6 +41,7 @@ include("filters.jl")
 include("formatters.jl")
 include("handlers.jl")
 include("loggers.jl")
+include("deprecates.jl")
```
Comment (omus) [inline src/Memento.jl]: I think either the file name "deprecated.jl" or "deprecations.jl" would read better.

File: src/deprecates.jl [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+Handles printing any records with any `Formatter` and a `Syslog` `IO` type.
+"""
+function Memento.emit(handler::DefaultHandler{F, O}, rec::Record) where {F<:Formatter, O<:Syslogs.Syslog}
+    Base.depwarn("The custom `Memento.emit` method for Syslog IO types will not be provided in the future.", :emit)
+    println(handler.io, rec[:level], format(handler.fmt, rec))
+    flush(handler.io)
+end
+
+@deprecate JsonFormatter(aliases=Nullable()) DictFormatter(aliases, JSON.json)
```
Comment (omus) [inline src/deprecates.jl]: I recommend putting comment fences around the deprecations so it is easier to keep track of when the should be removed. See: https://github.com/invenia/Mocking.jl/blob/master/src/deprecated.jl

File: test/deprecates.jl [hunk 3]
```diff
@@ -0,0 +1,37 @@
+@testset "Deprecations" begin
+    @testset "Syslog" begin
+        if Sys.isunix()
```
Comment (omus) [inline test/deprecates.jl]: Maybe have the `if` surround the testset so that this doesn't show up in the final count?

File: test/deprecates.jl [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+            levels = copy(Memento._log_levels)
+            levels["invalid"] = 100
+
+            # Create our DefaultHandler w/ the Syslog IO type
+            handler = DefaultHandler(Memento.Syslog(), DefaultFormatter("{level}: {msg}"))
```
Comment (omus) [inline test/deprecates.jl]: FYI:
```julia
julia> Array{Int}(1)
WARNING: Array{T}(m::Int) where T is deprecated, use Array{T}(uninitialized, m) instead.
Stacktrace:
 [1] depwarn at ./deprecated.jl:68 [inlined]
 [2] Array{Int64,N} where N(::Int64) at ./deprecated.jl:56
 [3] top-level scope
 [4] eval(::Module, ::Expr) at ./repl/REPL.jl:3
 [5] eval_user_input(::Any, ::Base.REPL.REPLBackend) at ./repl/REPL.jl:69
 [6] macro expansion at ./repl/REPL.jl:100 [inlined]
 [7] (::getfield(Base.REPL, Symbol("##1#2")){Base.REPL.REPLBackend})() at ./event.jl:95
in expression starting at no file:0
1-element Array{Int64,1}:
 4596391456
```

```julia
julia> using Test

julia> @test_warn "Array{T}(m::Int) where T is deprecated, use Array{T}(uninitialized, m) instead." Array{Int}(1)
1-element Array{Int64,1}:
 4520842800
```
Unfortunately I think this is only on Julia 0.7

Comment (rofinn) [reply test/deprecates.jl]: I should be able to use `@test_warn` with `Compat.Test`.

File: src/Memento.jl (hunk 1 above)
Comment (rofinn) [reply src/Memento.jl]: "deprecates" is what a lot of the stats libraries use, but I have no strong preference.

**Pull request merged**